diff --git a/buildscripts/include/download-deps.sh b/buildscripts/include/download-deps.sh
index 64bdc65..6773863 100644
--- a/buildscripts/include/download-deps.sh
+++ b/buildscripts/include/download-deps.sh
@@ -15,7 +15,7 @@ if [ ! -d mbedtls ]; then
 fi
 
 # dav1d
-[ ! -d dav1d ] && git clone https://github.com/videolan/dav1d
+[ ! -d dav1d ] && git clone --depth=1 --branch 1.5.4 https://github.com/videolan/dav1d
 
 # ffmpeg
 if [ ! -d ffmpeg ]; then
@@ -46,7 +46,7 @@ if [ ! -d fontconfig ]; then
 fi
 
 # libass
-[ ! -d libass ] && git clone https://github.com/libass/libass
+[ ! -d libass ] && git clone --depth=1 --branch 0.17.5 https://github.com/libass/libass
 
 # lua
 if [ ! -d lua ]; then
@@ -56,7 +56,7 @@ if [ ! -d lua ]; then
 fi
 
 # libplacebo
-[ ! -d libplacebo ] && git clone --recursive https://github.com/haasn/libplacebo
+[ ! -d libplacebo ] && git clone --depth=1 --branch v7.360.1 --recursive --shallow-submodules https://github.com/haasn/libplacebo
 
 # curl
 if [ ! -d curl ]; then
@@ -67,6 +67,6 @@ if [ ! -d curl ]; then
 fi
 
 # mpv
-[ ! -d mpv ] && git clone https://github.com/mpv-player/mpv
+[ ! -d mpv ] && git clone --depth=1 --branch v0.41.0 https://github.com/mpv-player/mpv
 
 cd ..
diff --git a/buildscripts/include/download-sdk.sh b/buildscripts/include/download-sdk.sh
index 7a61283..cc300b4 100644
--- a/buildscripts/include/download-sdk.sh
+++ b/buildscripts/include/download-sdk.sh
@@ -75,9 +75,9 @@ elif [ -z "${os_ndk}" ]; then
 	echo y | sdkmanager "ndk;${v_ndk_n}"
 	ln -s "android-sdk-$os/ndk/${v_ndk_n}" "android-ndk-${v_ndk}"
 else
 	echo "Downloading NDK."
-	$WGET "http://dl.google.com/android/repository/android-ndk-${v_ndk}-${os_ndk}.zip"
+	$WGET "https://dl.google.com/android/repository/android-ndk-${v_ndk}-${os_ndk}.zip"
 	unzip -q "android-ndk-${v_ndk}-${os_ndk}.zip"
 	rm "android-ndk-${v_ndk}-${os_ndk}.zip"
 fi
 if ! grep -qF "${v_ndk_n}" "android-ndk-${v_ndk}/source.properties"; then
@@ -96,7 +96,7 @@ fi
 
 # gas-preprocessor
 mkdir -p bin
-$WGET "https://github.com/FFmpeg/gas-preprocessor/raw/master/gas-preprocessor.pl" \
+$WGET "https://github.com/FFmpeg/gas-preprocessor/raw/ac1836309c2e77023c228b7184485597286289d3/gas-preprocessor.pl" \
 	-O bin/gas-preprocessor.pl
 chmod +x bin/gas-preprocessor.pl
 
diff --git a/buildscripts/include/ci.sh b/buildscripts/include/ci.sh
--- a/buildscripts/include/ci.sh
+++ b/buildscripts/include/ci.sh
@@ -63,1 +63,1 @@
-	$WGET https://github.com/mpv-player/mpv/archive/master.tar.gz -O master.tgz
+	$WGET https://github.com/mpv-player/mpv/archive/41f6a645068483470267271e1d09966ca3b9f413.tar.gz -O master.tgz
diff --git a/buildscripts/scripts/ffmpeg.sh b/buildscripts/scripts/ffmpeg.sh
index e920849..bfc3d48 100755
--- a/buildscripts/scripts/ffmpeg.sh
+++ b/buildscripts/scripts/ffmpeg.sh
@@ -29,7 +29,10 @@ args=(
 	--extra-cflags="-I$prefix_dir/include $cpuflags" --extra-ldflags="-L$prefix_dir/lib"
 
 	--enable-{jni,mediacodec,mbedtls,libdav1d,libxml2} --disable-vulkan
-	--disable-static --enable-shared --enable-{gpl,version3}
+	--disable-static --enable-shared
+	--disable-gpl --disable-nonfree --enable-version3
+	--enable-filter=loudnorm,dynaudnorm,acompressor,alimiter,volume,aresample,ebur128,astats
+	--enable-swresample
 
 	# disable unneeded parts
 	--disable-{stripping,doc,programs}
diff --git a/buildscripts/scripts/mpv.sh b/buildscripts/scripts/mpv.sh
index 647f146..f780e85 100755
--- a/buildscripts/scripts/mpv.sh
+++ b/buildscripts/scripts/mpv.sh
@@ -18,8 +18,9 @@ unset CC CXX # meson wants these unset
 meson setup $build --cross-file "$prefix_dir"/crossfile.txt \
 	--default-library shared \
 	-Diconv=disabled -D{lua,libcurl}=enabled \
+	-Dgpl=false \
 	-Dlibmpv=true -Dcplayer=false \
-	-Dmanpage-build=disabled
+	-Dmanpage-build=disabled -Dbuild-date=false
 
 ninja -C $build -j$cores
 if [ -f $build/libmpv.a ]; then
