diff --git a/scripts/ffmpeg-config b/scripts/ffmpeg-config
index d9bc544..13ac24c 100755
--- a/scripts/ffmpeg-config
+++ b/scripts/ffmpeg-config
@@ -11,7 +11,7 @@ if test -f "$BUILD"/ffmpeg_options ; then
     unset -v IFS
 fi
 
-OPTIONS="--enable-gpl --disable-debug --disable-doc --enable-static --disable-shared --enable-pic"
+OPTIONS="--disable-gpl --disable-nonfree --enable-version3 --disable-debug --disable-doc --enable-static --disable-shared --enable-pic"
 
 # Do FFmpeg's job.
 if ( echo "$OPTIONS" "$@" | \
@@ -22,8 +22,8 @@ then
     if pkg-config gnutls ; then
         OPTIONS="$OPTIONS --enable-gnutls"
         echo "Auto-enabling GnuTLS."
     elif pkg-config openssl ; then
-        OPTIONS="$OPTIONS --enable-nonfree --enable-openssl"
-        echo "Auto-enabling OpenSSL (creates a non-redistributable binary)."
+        OPTIONS="$OPTIONS --enable-openssl"
+        echo "Auto-enabling OpenSSL for the LGPLv3 runtime."
     fi
 fi
